crates/core_simd/src/swizzle_dyn.rs

```diff
@@ -29,13 +29,17 @@ where
         // SAFETY: Intrinsics covered by cfg
         unsafe {
             match N {
-                #[cfg(target_feature = "neon")]
+                #[cfg(all(target_feature = "neon", target_endian = "little"))]
```

Comment:
I'm not sure if this is what @workingjubilee was referring to, but I think it needs to be this:

```suggestion
                #[cfg(all(
                    any(
                        target_arch = "aarch64",
                        all(target_arch = "arm", target_feature = "v7")
                    ),
                    target_feature = "neon",
                    target_endian = "little"
                ))]
```

Reply:
I've applied your suggestion.